web_widget_radio_tree/static/src/js/widget.js

```diff
@@ -0,0 +1,90 @@
+openerp.web_widget_radio_tree = function (instance) {
+
+    var QWeb = instance.web.qweb;
+
+    instance.web.list.columns.add('field.radio_tree', 'instance.web.list.RadioTreeColumn');
+
+    instance.web.list.RadioTreeColumn = instance.web.list.Column.extend({
+        _format: function (row_data, options) {
+            var name = options.model + '_' + this.id;
+            return _.template(
+                '<input type="radio" name="<%-name%>" <%-checked%> readonly="readonly"></input>', {
+                    name: name,
+                    checked: row_data[this.id].value ? 'checked' : '',
+                });
+        }
+    });
+
+    instance.web.form.widgets.add('radio_tree', 'instance.web.form.RadioTree');
+
+    instance.web.form.RadioTree = instance.web.form.FieldBoolean.extend({
+        template: 'RadioTree',
+        start: function() {
+            var self = this;
+            this.$checkbox = $('input', this.$el);
+            var radio_name = this.__parentedParent.model + '_' + this.$checkbox[0].name;
+            this.$checkbox.attr('name', radio_name);
+            this.setupFocus(this.$checkbox);
+            this.$el.click(_.bind(function() {
+                self.clean_radio_in_records();
+                this.internal_set_value(true);
+            }, this));
+            var check_readonly = function() {
+                self.$checkbox.prop('disabled', self.get('effective_readonly'));
+                self.click_disabled_boolean();
+            };
+            this.on('change:effective_readonly', this, check_readonly);
+            check_readonly.call(this);
+            this._super.apply(this, arguments);
+        },
+        click_disabled_boolean: function(){
+            var $disabled = this.$el.find('input[type=radio]:disabled');
+            $disabled.each(function (){
+                $(this).next('div').remove();
+                $(this).closest('span').append($('<div class="boolean"></div>'));
+            });
+        },
+        clean_radio_in_records: function() {
```

Comment:
ah, I see. Then I think you do the right thing in principle, but it could be a bit more elegant in my opinion:
- use https://github.com/OCA/OCB/blob/8.0/addons/web/static/src/js/openerpframework.js#L229 instead of the private field
- you use properties specific to `BufferedDataSet`, this way, your widget is not going to work with many2many fields. Better read all ids and call `write`, the implementation of the specific dataset will take care to do the right thing hopefully